crates/storage-pg/src/user/mod.rs

```diff
@@ -53,16 +54,29 @@ impl<'c> PgUserRepository<'c> {
     }
 }
 
-#[derive(Debug, Clone)]
-struct UserLookup {
-    user_id: Uuid,
-    username: String,
-    primary_user_email_id: Option<Uuid>,
-    created_at: DateTime<Utc>,
-    locked_at: Option<DateTime<Utc>>,
-    can_request_admin: bool,
+mod priv_ {
+    // The enum_def macro generates a public enum, which we don't want, because it
+    // triggers the missing docs warning
+    #![allow(missing_docs)]
+
+    use chrono::{DateTime, Utc};
+    use sea_query::enum_def;
+    use uuid::Uuid;
+
+    #[derive(Debug, Clone, sqlx::FromRow)]
+    #[enum_def]
```

Comment:
can you comment what this macro does? It isn't documented upstream :/: https://docs.rs/sea-query/latest/sea_query/attr.enum_def.html

It doesn't even make intuitive sense, this is a struct not an enum...?

Reply:
This is documented here: https://docs.rs/sea-query/latest/sea_query/index.html#iden

Basically it creates an enumeration with 'names' for each field, so that you get in this case `UserLookupIden::UserId`, `UserLookupIden::Username`, etc.

This is used then in the query builder later to make sure we reference actual columns in this struct